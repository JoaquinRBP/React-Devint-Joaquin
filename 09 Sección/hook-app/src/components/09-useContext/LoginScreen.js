import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const LoginScreen = () => {
    const {user,setuser} = useContext(UserContext);
    const usuarioLogin ={
        id: 8,
        username: 'Joaquin',
        password: '123456'
    }
return (
    <>
        <h1>LoginScreen</h1>
        <hr />

        <pre>
            {JSON.stringify(user,null, 3)}
        </pre>
        <button 
        className="btn btn-primary"
        onClick={ ()=> setuser(usuarioLogin)}>
            Login
        </button>
    </>
)
}
