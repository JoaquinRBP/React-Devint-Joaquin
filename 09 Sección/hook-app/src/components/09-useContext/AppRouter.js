import React from 'react'
import {
    BrowserRouter as Router,
    Route,
    Routes
    } from "react-router-dom";
import {AboutScreen} from "./AboutScreen";
import {LoginScreen} from "./LoginScreen";
import {HomeScreen}  from "./HomeScreen";
import {NavBar} from "./NavBar";
export const AppRouter = () => {
    //Routes vs Switch
    //Element vs Component
    //Redirect vs NoMatch
return (
    <Router>
        <div>
            <NavBar />
            <div className="container">
            <Routes>
                <Route exact path="/" element={<HomeScreen />} />
                <Route exact path="/about" element={<AboutScreen />} />
                <Route exact path="/login" element={<LoginScreen />} />
                <Route path='*' element={<main>
                    <h1>ERR 404</h1>
                </main>} />
            </Routes>
            </div>
        </div>
    </Router>
)
}
