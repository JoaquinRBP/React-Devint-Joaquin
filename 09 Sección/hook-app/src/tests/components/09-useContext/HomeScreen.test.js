import { mount, shallow } from 'enzyme';
import React from 'react';
import {HomeScreen} from '../../../components/09-useContext/HomeScreen';
import {UserContext} from '../../../components/09-useContext/UserContext';
describe('Pruebas sobre HomeScreen', () => {
    const user = {
        name: 'Joaquin',
        email: 'joaquin@gmail.com'
    }
    const wrapper = mount(
    <UserContext.Provider
        value={user}
    >
        <HomeScreen/>
    </UserContext.Provider>);
    test('Debereia mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
})
