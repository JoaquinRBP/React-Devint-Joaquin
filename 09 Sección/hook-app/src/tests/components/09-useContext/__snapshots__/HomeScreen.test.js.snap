// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre HomeScreen Debereia mostrarse correctamente 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre />,
]
`;
